### PR TITLE
test: add task queue ut

### PR DESCRIPTION
**Pull request by joeylichang**
### Description

Add task queue ut.

### Rationale

N/A

### Example

N/A

### Changes

Notable changes: 

N/A

### CODE REVIEW

**Review by will-2012 — APPROVED** [review]

File: base/gfsptqueue/queue_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/bnb-chain/greenfield-storage-provider/base/types/gfsptask"
+	"github.com/bnb-chain/greenfield-storage-provider/core/task"
+)
+
+func TestApprovalTaskRetireByExpiredHeight(t *testing.T) {
+	task1 := &gfsptask.GfSpCreateObjectApprovalTask{
+		CreateObjectInfo: &storagetypes.MsgCreateObject{
+			ObjectName:        "test_task_1",
```
Comment (sysvm) [inline base/gfsptqueue/queue_test.go:16]: moved these unit test to testCases struct may be better.

File: base/gfsptqueue/queue_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestReplicateTaskRetireByExpired(t *testing.T) {
+	task1 := &gfsptask.GfSpReplicatePieceTask{
```
Comment (sysvm) [inline base/gfsptqueue/queue_test.go:109]: ditto

File: base/gfsptqueue/queue_test.go (hunk 1 above)
Comment (joeylichang) [reply base/gfsptqueue/queue_test.go:16]: the task pointer is be used, inner struct can not accquie the pointer

File: base/gfsptqueue/queue_test.go (hunk 2 above)
Comment (joeylichang) [reply base/gfsptqueue/queue_test.go:109]: ditto

**Pull request merged**